eva-accession-import-automation/validate_imported_contigs_for_assembly.py

```diff
@@ -0,0 +1,132 @@
+        config = dict(parser.items(section=properties_section_name))
+        return config
+
+
+def validate_imported_contigs(assembly_properties_file, config_file):
+    config = get_args_from_private_config_file(config_file)
+    config.update(get_args_from_assembly_properties_file(assembly_properties_file))
+    config["assembly_report_path"] = config["parameters.assemblyReportUrl"].split("file:")[-1]
+    config["assembly_md5"] = hashlib.md5(config["parameters.assemblyName"].encode("utf-8")).hexdigest()
+    # We need these because string interpolation won't work if there is a dot in the key
+    config["taxonomy_accession"] = config["parameters.taxonomyAccession"]
+
+    config["contig_chr_mismatch_table"] = "dbsnp_ensembl_species.dbsnp_species_with_contig_chromosome_start_mismatch"
+    config["contig_chr_match_table"] = "dbsnp_ensembl_species.dbsnp_species_with_contig_chromosome_start_match"
+    final_formatting_genbank_accessions_cmd = r"cut -d$'\t' -f5 | sort | uniq | paste -s - | sed s/$'\t'/\",\"/g | sed s/$/\"/g | sed s/^/\"/g"
+    get_contigs_start_mismatch_cmd = "psql -A -t -h {metahost} -U {metauser} -d {metadb} -c " \
+                                     "\"select distinct contig_name from {contig_chr_mismatch_table} " \
+                                     "where table_name like '%{assembly_md5}%'" \
+                                     " and schema_name like '%{taxonomy_accession}%'\" -P pager=off " \
+                                     "| grep {assembly_report_path} -f - |".format(**config) + \
+                                     final_formatting_genbank_accessions_cmd
+    get_contigs_start_match_cmd = "psql -A -t -h {metahost} -U {metauser} -d {metadb} -c " \
+                                  "\"select distinct contig_name from {contig_chr_match_table} " \
+                                  "where table_name like '%{assembly_md5}%'" \
+                                  " and schema_name like '%{taxonomy_accession}%'\" -P pager=off " \
+                                  "| grep {assembly_report_path} -f - | grep -v assembled-molecule |".format(**config) \
+                                  + final_formatting_genbank_accessions_cmd
+    mongo_run_command_template = "mongo --quiet --host {0} --port {1} --username {2} " \
+                                 "--password {3} --authenticationDatabase=admin " \
+                                 "{4} --eval 'db.dbsnpSubmittedVariantEntity.findOne({{\"seq\": \"{5}\", " \
+                                 "\"contig\": {{$in: [{6}]}}}})'"
+
+    mismatch_contig_set = run_command_with_output("Get contigs with start mismatch against chromosome:",
+                                                  get_contigs_start_mismatch_cmd)[1].strip()
+    match_contig_set = run_command_with_output("Get contigs with start match against chromosome:",
+                                               get_contigs_start_match_cmd)[1].strip()
+
+    if mismatch_contig_set != '""':
```

Comment:
The idiom in Python does not need the comparison:
```suggestion
    if mismatch_contig_set:
```

Reply:
This won't work because what we are comparing against (the output from the sed command) is not an empty string but a literal set of double quotes.